### PR TITLE
Enable dependabot version updates for Cargo

### LINKED ISSUE #30: Enable Dependabot
While we have Dependabot enabled, it is not configured, so it's not doing anything for us. We should add a configuration file to allow it to do its thing.

### PR DESCRIPTION
**Pull request by mbestavros**
As discussed in today's standup. It's currently set to run daily.

Resolves #30.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+  - package-ecosystem: "cargo"
+    directory: "/"
+    # Check the npm registry for updates every day (weekdays)
```
Comment (connorkuehl) [inline .github/dependabot.yml:5]: ~~`# Check the npm registry for updates every day (weekdays)`~~

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+  - package-ecosystem: "cargo"
+    directory: "/"
```
Comment (connorkuehl) [inline .github/dependabot.yml:4]: We need this to also run for the crates under `internal/`

**Review by connorkuehl — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 2 above)
Comment (npmccallum) [reply .github/dependabot.yml:4]: To be clear, this is the only repo that has this problem.

File: .github/dependabot.yml (hunk 1 above)
Comment (mbestavros) [reply .github/dependabot.yml]: Whoops. Fixed, thanks!

File: .github/dependabot.yml (hunk 2 above)
Comment (mbestavros) [reply .github/dependabot.yml]: I've just added a second `cargo` entry with a second directory path. I think that's how you're supposed to do it.

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+version: 2
+updates:
+    - package-ecosystem: "cargo"
+        directory: "/"
+        schedule:
+            interval: "daily"
+    - package-ecosystem: "cargo"
+        directory: "internal/"
+        schedule:
+        interval: "daily"
```
Comment (npmccallum) [inline .github/dependabot.yml:10]: This is still not correct.

```suggestion
    - package-ecosystem: "cargo"
        directory: "internal/rcrt1"
        schedule:
            interval: "daily"
    - package-ecosystem: "cargo"
        directory: "internal/sallyport"
        schedule:
            interval: "daily"
    - package-ecosystem: "cargo"
        directory: "internal/shim-sev"
        schedule:
            interval: "daily"
    - package-ecosystem: "cargo"
        directory: "internal/shim-sgx"
        schedule:
            interval: "daily"
```

**Review by npmccallum — CHANGES_REQUESTED** [review]

**Review by npmccallum — APPROVED** [review]

**Review by connorkuehl — APPROVED** [review]

**Pull request merged**